tests/functional/test_notebooks.py

```diff
@@ -46,277 +46,508 @@ def generator(self):
+        "--defaultEntrypoint", "some_entrypoint",
+        "--containerUser", "some_container_user",
+        "--shutdownTimeout", "8",
+        "--isPreemptible",
+    ]
+    EXPECTED_REQUEST_JSON_WITH_ALL_OPTIONS = {
+        "vmTypeId": 25,
+        "containerId": 123,
+        "clusterId": "321",
+        "name": "some_notebook_name",
+        "registryUsername": "some_username",
+        "registryPassword": "some_password",
+        "defaultEntrypoint": "some_entrypoint",
+        "containerUser": "some_container_user",
+        "shutdownTimeout": 8,
+        "isPreemptible": True,
+        "isPublic": False,
+    }
+    COMMAND_WITH_OPTIONS_FILE_USED = ["notebooks", "create", "--optionsFile", ]  # path added in test
+
+    RESPONSE_JSON_WITH_WRONG_API_TOKEN = {"status": 400, "message": "Invalid API token"}
+    EXPECTED_STDOUT_WITH_WRONG_API_TOKEN = "Failed to create resource: Invalid API token\n"
+
+    @mock.patch("gradient.api_sdk.clients.http_client.requests.get")
+    @mock.patch("gradient.api_sdk.clients.http_client.requests.post")
+    def test_should_send_post_request_and_print_notebook_id(self, post_patched, get_patched):
+        post_patched.return_value = MockResponse(self.EXPECTED_RESPONSE_JSON)
+        get_patched.return_value = MockResponse(example_responses.NOTEBOOK_GET_RESPONSE)
+
+        runner = CliRunner()
+        result = runner.invoke(cli.cli, self.COMMAND)
+
+        assert result.output == self.EXPECTED_STDOUT, result.exc_info
+        post_patched.assert_called_once_with(self.URL,
+                                             headers=EXPECTED_HEADERS,
+                                             json=self.EXPECTED_REQUEST_JSON,
+                                             data=None,
+                                             files=None,
+                                             params=None)
+        assert EXPECTED_HEADERS["X-API-Key"] != "some_key"
+
+    @mock.patch("gradient.api_sdk.clients.http_client.requests.get")
+    @mock.patch("gradient.api_sdk.clients.http_client.requests.post")
+    def test_should_send_changed_headers_when_api_key_option_was_used(self, post_patched, get_patched):
+        post_patched.return_value = MockResponse(self.EXPECTED_RESPONSE_JSON)
+        get_patched.return_value = MockResponse(example_responses.NOTEBOOK_GET_RESPONSE)
+
+        runner = CliRunner()
+        result = runner.invoke(cli.cli, self.COMMAND_WITH_API_KEY_USED)
+
+        assert result.output == self.EXPECTED_STDOUT, result.exc_info
+        post_patched.assert_called_once_with(self.URL,
+                                             headers=EXPECTED_HEADERS_WITH_CHANGED_API_KEY,
+                                             json=self.EXPECTED_REQUEST_JSON,
+                                             data=None,
+                                             files=None,
+                                             params=None)
+        assert EXPECTED_HEADERS["X-API-Key"] != "some_key"
```

Comment:
what this line check actually?

Reply:
Not sure we just uncommented it. I'll remove these assertions.

Reply:
turns out this line occurs dozens of times in different tests and I have no idea what was it suppose to do :D I'll check it with @BartoszCki later and probably we'll get rid of it